<?xml version="1.0"?>
<launch>

  <!-- <param name="/use_sim_time" value="true"/> -->

  <!-- <arg name="xmin" default="-100.0"/> -->
  <!-- <arg name="ymin" default="-100.0"/> -->
  <!-- <arg name="xmax" default="100.0"/> -->
  <!-- <arg name="ymax" default="100.0"/> -->

  <arg name="site_name" default="riseholme"/>
  <arg name="datum_file" default="$(find mel_amcl)/config/datums/$(arg site_name).yaml"/>

  <rosparam command="load" file="$(arg datum_file)" />  

  <node pkg="gmapping" type="slam_gmapping" name="gmapping" output="screen" respawn="true">
    <param name="base_frame" value="base_link"/>
    <param name="odom_frame" value="odom"/>
    <param name="map_update_interval" value="15.0"/>
    <param name="minUrange" value="1.5"/>
    <param name="maxUrange" value="6.0"/> <!-- 6.5 -->
    <param name="maxRange" value="30.0"/>
    <param name="sigma" value="0.1"/>
    <param name="kernelSize" value="0.05"/>
    <param name="lstep" value="0.005"/>
    <param name="astep" value="0.005"/>
    <param name="iterations" value="2"/>
    <param name="lsigma" value="0.08"/>
    <param name="ogain" value="15.0"/>
    <param name="lskip" value="0"/>
    <param name="minimumScore" value="99999"/>
    <param name="srr" value="0.000001"/>
    <param name="srt" value="0.000002"/>
    <param name="str" value="0.000001"/>
    <param name="stt" value="0.000002"/>
    <param name="linearUpdate" value="0.5"/>
    <param name="angularUpdate" value="0.436"/>
    <param name="temporalUpdate" value="-1.0"/>
    <param name="resampleThreshold" value="0.5"/>
    <param name="particles" value="80"/>


    <!-- <param name="xmin" value="$(arg xmin)"/> -->
    <!-- <param name="ymin" value="$(arg ymin)"/> -->
    <!-- <param name="xmax" value="$(arg xmax)"/> -->
    <!-- <param name="ymax" value="$(arg ymax)"/> -->
    <param name="delta" value="0.05"/>



    <param name="llsamplerange" value="0.02"/>
    <param name="llsamplestep" value="0.02"/>
    <param name="lasamplerange" value="0.02"/>
    <param name="lasamplestep" value="0.02"/>


    <param name="occ_thresh" value="0.25"/>


  </node>

</launch>
